src/basic/capability-util.c

```diff
@@ -335,7 +341,7 @@ int drop_privileges(uid_t uid, gid_t gid, uint64_t keep_capabilities) {
                     cap_set_flag(d, CAP_PERMITTED, j, bits, CAP_SET) < 0)
                         return log_error_errno(errno, "Failed to enable capabilities bits: %m");
 
-                if (cap_set_proc(d) < 0)
+                if (cap_set_proc(d) < 0 && (errno != EPERM || (detect_container() <= 0)))
```

Comment:
AFAICS, we don't really check that we have the capabilities listed in `keep_capabilities`, so that cap_set_proc adds more capabilties than we originally had.  Anyway, I am going to drop this change and require the container to have CAP_NEW_RAW and CAP_NET_BROADCAST, as required by systemd-networkd.